data/src/main/java/com/linkedin/data/schema/annotation/SchemaAnnotationHandler.java

```diff
@@ -144,6 +149,93 @@ default SchemaVisitor getVisitor()
     return new PathSpecBasedSchemaAnnotationVisitor(this);
   }
 
+  /**
+   * Check annotations changes are compatible or not
+   *
+   * @param olderResolvedProperties the older resolvedProperties
+   * @param newerResolvedProperties the newer resolvedProperties
+   * @param olderContext the older annotationCheck context
+   * @param newerContext the newer annotationCheck context
+   * @return AnnotationCompatibilityResult
+   */
+  default AnnotationCompatibilityResult annotationCompatibilityCheck(Map<String,Object> olderResolvedProperties, Map<String, Object> newerResolvedProperties,
+      AnnotationCheckContext olderContext, AnnotationCheckContext newerContext)
+  {
+    return new AnnotationCompatibilityResult();
```

Comment:
If we make it default method, that means every handler will return a `AnnotationCompatibilityResult`. Is there a way to tell whether or not this check happened? How can we distinguish the two cases

Reply:
We would set up null as default value here to differentiate whether this method is conducted or not. But using null as default value is not ideal, I think adding a flag is a cleaner way to achieve it.